### PR TITLE
cylc lint: review feedback

### PR DESCRIPTION
**Pull request by oliver-sanders**
@wxtim - Quick fixes for two small issues and some suggested changes.

### CODE REVIEW

File: cylc/flow/scripts/lint.py [hunk 1]
```diff
@@ -14,19 +14,18 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Cylc configuration linter.
+
+Checks code style, deprecated syntax and other issues."""
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:19]: The docstring is functional for `cylc help all`, just need to give it two lines to follow the pattern of other commands.

File: cylc/flow/scripts/lint.py [hunk 2]
```diff
@@ -14,19 +14,18 @@
...
+
 COP_DOC = """cylc lint [OPTIONS] ARGS
 
 Cylc lint looks through one or more folders for ".cylc" and ".rc"
 files.
 
-By default it will examine files for code style.
-If a workflow has been upgraded to Cylc 8 by changing the top level
-"suite.rc" to "flow.cylc" then it will also check for deprecated
-Cylc 7 syntax.
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:25]: I think we can make this less verbose. The automatic rule selection is sufficiently clean that I don't think we need to explain it (we don't for other commands).

File: cylc/flow/scripts/lint.py [hunk 3]
```diff
@@ -14,19 +14,18 @@
...
-
-You can run "cylc lint" with each set of rules alone using
-"cylc lint -r style" (for the style checks) or  "cylc lint -r 728"
-(for the Cylc 7 syntax checks).
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:29]: I don't think turning off rulesets is a large use case so the option docs should be enough.

File: cylc/flow/scripts/lint.py [hunk 4]
```diff
@@ -36,17 +35,16 @@
...
-# run inplace
-cylc lint --inplace <paths to workflow directories to check>
-cylc lint -i <paths to workflow directories to check>
-
-# Get information about errors:
-cylc lint --reference
-cylc lint -R
+  # Get information about errors:
+  $ cylc lint --reference
+  $ cylc lint -R
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:47]: Indent and format to match other commands.

File: cylc/flow/scripts/lint.py [hunk 5]
```diff
@@ -326,7 +326,7 @@ def check_cylc_file(file_, checks, modify=False):
                     print(
                         Fore.YELLOW +
                         f'[{message["purpose"]}{message["index"]:03d}]'
-                        f'{file_}: {line_no}: {message["short"]}'
+                        f' {file_rel}:{line_no}: {message["short"]}'
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:329]: Use relative filepaths in reporting (ala flake8) to make the summary easier to read.

File: cylc/flow/scripts/lint.py [hunk 6]
```diff
@@ -443,9 +442,9 @@ def get_option_parser() -> COP:
         dest='linter'
     )
     parser.add_option(
-        '--reference', '--ref', '-R',
+        '--list-codes',
         help=(
-            'Print Reference for error codes.'
+            'List all linter codes.'
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:447]: Having both `-r` and `-R` as options seems confusing. I don't think we need a short option for a documentation command. How about `--list-codes`?

File: cylc/flow/scripts/lint.py [hunk 7]
```diff
@@ -469,6 +467,10 @@ def main(parser: COP, options: 'Values', *targets) -> None:
     else:
         targets = (str(Path.cwd()),)
 
+    # make sure the targets are all src/run directories
+    for target in targets:
+        check_flow_file(target)
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:472]: Without this the linter will attempt to recursively glob for Cylc files even in directories which are not workflow directories.

Discovered when I typed `cylc lint` in my home dir.

File: cylc/flow/scripts/lint.py [hunk 8]
```diff
@@ -501,23 +499,36 @@ def main(parser: COP, options: 'Values', *targets) -> None:
...
-
         if options.linter == 'all':
             checks_done = "728 & style"
         else:
             checks_done = options.linter
-        msg = (
-            f'Checked {target} against {checks_done} '
-            f'rules and found {count} issues.'
-        )
-        print(f'{color}{"-" * len(msg)}\n{msg}')
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:520]: The `--------` line didn't fit in my terminal so I got a line break. I think we can just use a newline here for a cleaner look?

File: cylc/flow/scripts/lint.py [hunk 9]
```diff
@@ -501,23 +499,36 @@ def main(parser: COP, options: 'Values', *targets) -> None:
...
+
+        if count > 0:
+            msg = (
+                f'\n{Fore.YELLOW}'
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:517]: Only print the newline if there was something to report, avoids a blank line for a workflow with no lint.

File: cylc/flow/scripts/lint.py [hunk 10]
```diff
@@ -501,23 +499,36 @@ def main(parser: COP, options: 'Values', *targets) -> None:
...
+                f'Checked {target} against {checks_done} '
+                f'rules and found {count} issue'
+                f'{"s" if count > 1 else ""}.'
+            )
+        else:
+            msg = (
+                f'{Fore.GREEN}'
+                f'Checked {target} against {checks_done} '
+                'Found no issues.'
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:526]: Slightly simpler message in the "nothing to report" case.

File: cylc/flow/scripts/lint.py [hunk 11]
```diff
@@ -489,10 +491,6 @@ def main(parser: COP, options: 'Values', *targets) -> None:
                 )
                 continue
             elif not cylc8 and options.linter == 'all':
-                LOG.warn(
-                    f'{target} not a Cylc 8 workflow: '
-                    'Running style checks only...'
-                )
```
Comment (oliver-sanders) [inline cylc/flow/scripts/lint.py:495]: No need for a scary warning here, the migration checks are not valid for this case, they can be silently dropped.

File: cylc/flow/scripts/lint.py (hunk 6 above)
Comment (wxtim) [reply cylc/flow/scripts/lint.py:447]: Legit - I've confused myself at least once.

**Review by wxtim — APPROVED** [review]

**Pull request merged**